addon/modifiers/key-macro.ts

```diff
@@ -0,0 +1,67 @@
+import Modifier from 'ember-modifier';
+import {inject as service} from '@ember/service';
+import {assert} from '@ember/debug';
+
+export type IKeyMacro = {} | null;
+
+export interface IKeyManager {
+  addMacro(args: any): IKeyMacro;
+  removeMacro(macro: any): void;
+}
+
+const isMac: boolean = window.navigator.platform === "MacIntel";
+export const cmdKey: string = isMac ? 'Meta' : 'Control';
+
+export default class KeyMacroModifier extends Modifier {
```

Comment:
Thoughts on making this an abstract class with the property `abstract keyEvent: string` defined which would enforce that it can't be used directly (must be inherited) and its sub class must define `keyEvent`?

Reply:
Good idea. Used the KeyEvent type rather than string